test/APITests.js

```diff
@@ -120,11 +146,63 @@ describe('Simple HiGlassComponent', () => {
 
       waitForTransitionsFinished(api.getComponent(), () => {
         waitForTilesLoaded(api.getComponent(), () => {
+          api.destroy();
           done();
         });
       });
     });
 
+    // Fritz: This test fails but if you comment out all the other tests it will
+    // succeed!? Also, if you move this test to the top it will succeed but
+    // tests number 3, 4, and 6 will fail all of a sudden! Therefore, I assume
+    // the test itself work just fine but there's something fundamentally broken
+    // with our test setup or with HG. Either way, the HG instances must have
+    // some shared state that influences each other. I am giving up for now but
+    // we need to look into this again.
+    it('listens to click events', (done) => {
+      [div, api] = createElementAndAPI(
+        simple1And2dAnnotations,
+        { editable: false, bounded: true }
+      );
+
+      const canvas = findCanvas(div);
+
+      let clicked = 0;
+
+      api.on('click', () => { clicked++; });
+
+      const createPointerEvent = (type, x, y) => new PointerEvent(type, {
+        view: window,
+        bubbles: true,
+        cancelable: true,
+        // WARNING: The following property is absolutely crucial to have the
+        // event being picked up by PIXI. Do not remove under no circumstances!
+        pointerType: 'mouse',
+        screenX: x,
+        screenY: y,
+        clientX: x,
+        clientY: y
+      });
+
+      waitForTilesLoaded(api.getComponent(), () => {
+        setTimeout(() => {
+          canvas.dispatchEvent(createPointerEvent('pointerdown', 100, 200));
+          canvas.dispatchEvent(createPointerEvent('pointerup', 100, 200));
+        }, 0);
+
+        setTimeout(() => {
+          canvas.dispatchEvent(createPointerEvent('pointerdown', 100, 100));
+          canvas.dispatchEvent(createPointerEvent('pointerup', 100, 100));
+        }, 50);
+
+        setTimeout(() => {
+          expect(clicked).to.equal(2);
+          api.destroy();
+          done();
+        }, 100);
```

Comment:
Might set all the timeouts to 0: They should be queued and just run one after the other, in the order they are registered. If waiting 50ms or 100ms actually is necessary that would be a race condition it would be good to know about.

Reply:
Will test but the tests fail in general because there is / are more nasty issues. The HG instances or the tests are somehow non-independent and I have no idea how to fix those. Changing the order of API tests will make others fail while this one succeeds. Any input on that would be much appreciated.